salt/modules/macdefaults.py

```diff
@@ -98,15 +223,275 @@ def delete(domain, key, user=None):
 
         salt '*' macdefaults.delete NSGlobalDomain ApplePersistence
 
+        salt '*' macdefaults.delete NSGlobalDomain key.with.dots key_separator='.'
+
     domain
         The name of the domain to delete from
 
     key
-        The key of the given domain to delete
+        The key of the given domain to delete.
+        It can be a nested key separated by `key_separator`.
+
+    key_separator
+        The separator to use when splitting the key into a list of keys.
+        If None, the key will not be split (Default).
```

Comment:
Needs `..versionadded:: 3008`

Reply:
Done in commit d9243837c283d6659df911e8c2c5a44d7ce125ff